src/ai/backend/common/events/dispatcher.py

```diff
@@ -420,71 +469,45 @@ async def _consume_loop(self) -> None:
                 await self.dispatch_consumers(
                     decoded_event_name,
                     AgentId(msg.payload[b"source"].decode()),
                     msgpack.unpackb(msg.payload[b"args"]),
+                    [done],
                 )
-                await self._msg_queue.done(msg.msg_id)
-                self._metric_observer.observe_event_success(
-                    event_type=event_type,
-                    duration=time.perf_counter() - start,
-                )
-            except Exception as e:
-                self._metric_observer.observe_event_failure(
-                    event_type=event_type,
-                    duration=time.perf_counter() - start,
-                    exception=e,
-                )
-                log.exception("EventDispatcher.consume(): unexpected-error")
-            except BaseException as e:
-                self._metric_observer.observe_event_failure(
-                    event_type=event_type,
-                    duration=time.perf_counter() - start,
-                    exception=e,
-                )
-                raise
+            except Exception:
+                # Exception is already handled in handle() method.
+                pass
 
     @preserve_termination_log
     async def _subscribe_loop(self) -> None:
         async for msg in self._msg_queue.subscribe_queue():  # type: ignore
             if self._closed:
                 return
-            event_type = "unknown"
-            start = time.perf_counter()
+            decoded_event_name = msg.payload[b"name"].decode()
             try:
-                decoded_event_name = msg.payload[b"name"].decode()
-                if decoded_event_name and isinstance(decoded_event_name, str):
-                    event_type = decoded_event_name
                 await self.dispatch_subscribers(
                     decoded_event_name,
                     AgentId(msg.payload[b"source"].decode()),
                     msgpack.unpackb(msg.payload[b"args"]),
                 )
-                self._metric_observer.observe_event_success(
-                    event_type=event_type,
-                    duration=time.perf_counter() - start,
-                )
-            except Exception as e:
-                self._metric_observer.observe_event_failure(
-                    event_type=event_type,
-                    duration=time.perf_counter() - start,
-                    exception=e,
-                )
-                log.exception("EventDispatcher.subscribe(): unexpected-error")
-            except BaseException as e:
-                self._metric_observer.observe_event_failure(
-                    event_type=event_type,
-                    duration=time.perf_counter() - start,
-                    exception=e,
-                )
-                raise
+            except Exception:
+                # Exception is already handled in handle() method.
+                pass
```

Comment:
Why add except block if you're just going to pass?

Reply:
I added this part because all execptions are handled in the `EventDispatcher._handle()` method. Updated to remove this exception skipping logic.